File: src/ch04/solution_1.py
from rich import print


def tribonacci(n: int) -> int:
    if n == 0:
        return 0
    elif n == 1:
        return 0
    elif n == 2:
        return 1
    return tribonacci(n=n-1) + tribonacci(n=n-2) + tribonacci(n=n-3)


if __name__ == '__main__':
    result: int = tribonacci(n=10)
    print(result)

File: src/ch04/solution_1.rs
fn tribonacci(n: i32) -> i32 {
    match n {
        0 => 0,
        1 => 0,
        2 => 1,
        _ => tribonacci(n-1) + tribonacci(n-2) + tribonacci(n-3),
    }
}


fn main() {
    let result: i32 = tribonacci(10);
    println!("{}", result);
}